.github/workflows/vast.yaml

```diff
@@ -25,7 +25,9 @@ env:
   CCACHE_SLOPPINESS: 'file_macro,time_macros'
   CCACHE_DIR: '${{ github.workspace }}/.ccache'
   CCACHE_COMPRESS: true
-  CCACHE_COMPRESSLEVEL: 9
+  # Zstd compression level; up to 19 is supported by all versions.
+  # https://ccache.dev/manual/latest.html#config_compression_level
+  CCACHE_COMPRESSLEVEL: 19
```

Comment:
I would imagine that there is a non-linear increase in runtime with higher levels. Can we do some local tests to see what cost-benefit we are getting ourselves into?

Reply:
We only have 5GB of cache total for the repository, and right now every run is using around 1.5 GB. We just need to play around with it to see what makes sense for our workflows, so I suggest we take the binary search approach. I have a gut feeling that 19 will be alright to use, though. I'm planning to re-run CI a few times here to be able to see the cache size and average run time for 100% cache hits—which is where the decompression penalty is the largest.

Reply:
After some re-runs, here are my observations: The worst-case runtime increase seems to be less than 3 minutes. The combined cache size for one CI run went down from ~1.5 GB to ~0.5 GB. I'd say this is definitely worth merging.

Let's talk on Monday about this within @tenzir/backend.

Reply:
If we have a little data on the time vs. space dimensions, I'm all for this.

The results seems promising: cache size down by 3x for 3 mins extra. I'd be curious to see how much lower we can go until the gains in space become no longer so pronounced.

Reply:
> I'd be curious to see how much lower we can go until the gains in space become no longer so pronounced.

Technically it depends on the Zstd version that Ccache was built against, so we'd need to check that for all entries in our build matrix. 19 is the highest we can safely use per the linked Ccache docs.

Reply:
> The library supports regular compression levels from 1 up to `ZSTD_maxCLevel()`, which is currently 22. Levels >= 20, labeled `--ultra`, should be used with caution, as they require more memory. The library also offers negative compression levels, which extend the range of speed vs. ratio preferences. The lower the level, the faster the speed (at the cost of compression).

From http://facebook.github.io/zstd/zstd_manual.html